Hi!

Thanks for the PR. I will gladly take this!

Can you fix the two CI failures and then this is good to go:

```
$ make lint
golangci-lint run
cli.go:71:2: ifElseChain: rewrite if-else to switch statement (gocritic)
	if length == numSubDeviceIDs {
	^
cli.go:70:2: variable 'length' is only used in the if-statement (cli.go:71:2); consider using short syntax (ifshort)
	length := len(args)
	^
Makefile:24: recipe for target 'lint' failed
make: *** [lint] Error 1

The command "make lint" exited with 2.
```

--- a/README.md
+++ b/README.md
@@ -15,6 +15,26 @@ To search for devices using the CLI, pass in either:
 Here are some examples:
 
 ```text
+$ pciids 1ed5
+1ed5:0100 - Moore Threads Technology Co.,Ltd MTT S10
+1ed5:0101 - Moore Threads Technology Co.,Ltd MTT S10
+1ed5:0102 - Moore Threads Technology Co.,Ltd MTT S30
+1ed5:0105 - Moore Threads Technology Co.,Ltd MTT S50
+1ed5:0106 - Moore Threads Technology Co.,Ltd MTT S60
+1ed5:0111 - Moore Threads Technology Co.,Ltd MTT S100
+1ed5:0121 - Moore Threads Technology Co.,Ltd MTT S1000M
+1ed5:0122 - Moore Threads Technology Co.,Ltd MTT S1000
+1ed5:0123 - Moore Threads Technology Co.,Ltd MTT S2000
+1ed5:01ff - Moore Threads Technology Co.,Ltd MTT HDMI/DP Audio
+1ed5:0201 - Moore Threads Technology Co.,Ltd G2D30
+1ed5:0202 - Moore Threads Technology Co.,Ltd G2D20
+1ed5:0203 - Moore Threads Technology Co.,Ltd G2D10
+1ed5:0211 - Moore Threads Technology Co.,Ltd G2D40
+1ed5:0221 - Moore Threads Technology Co.,Ltd G2S80
+1ed5:0222 - Moore Threads Technology Co.,Ltd G2S85
+1ed5:0223 - Moore Threads Technology Co.,Ltd G2S4
+1ed5:0251 - Moore Threads Technology Co.,Ltd G2N10
+1ed5:02ff - Moore Threads Technology Co.,Ltd MTT HDMI/DP Audio
 $ pciids 1d0f efa1
 1d0f:efa1 - Amazon.com, Inc. Elastic Fabric Adapter (EFA)
 $ pciids 10de 2206 10de 1467
@@ -104,6 +124,7 @@ Users can take advantage of various functions in their own code:
 * `All()`: returns a list of all PCI ID devices
 * `LatestFile()`: string of the latest PCI ID database
 * `Parse(string)`: parses a given PCI ID database string
+* `QueryVendor(vendorID)`: Searches for devices matching a vendor ID
 * `QueryDevice(vendorID, deviceID)`: Searches for devices matching a PCI ID
    pair
 * `QuerySubDevice(vendorID, deviceID, subVendorID, subDeviceID)`: Like

--- a/cli.go
+++ b/cli.go
@@ -22,11 +22,13 @@ var rootCmd = &cobra.Command{
 	Long: `Lookup vendor and device names using PCI IDs.
 
 To search for devices using the CLI, pass in either:
-  a) a pair of vendor and device PCI IDs
-  b) two pairs, vendor and device PCI IDs as well as sub-vendor and
+  a) vendor
+  b) a pair of vendor and device PCI IDs
+  c) two pairs, vendor and device PCI IDs as well as sub-vendor and
      sub-device PCI IDs:
 
 Examples:
+$ pciids 1ed5
 $ pciids 1d0f efa1
 $ pciids 10de 2206 10de 1467
 `,
@@ -47,14 +49,14 @@ func setup(cmd *cobra.Command, args []string) {
 	}
 }
 
-// Checks that there are exactly two arguments.
+// Checks that there are exactly 1, 2, or 4 arguments.
 func args(cmd *cobra.Command, args []string) error {
-	if len(args) != numDeviceIDs {
-		if len(args) != numSubDeviceIDs {
-			return errors.New(
-				"either two or four PCI IDs as arguments (e.g. 10de 1467) are required",
-			)
-		}
+	length := len(args)
+	if length != numVendorIDs && length != numDeviceIDs &&
+		length != numSubDeviceIDs {
+		return errors.New(
+			"invalid number of arguments, expected 1, 2, or 4 (e.g. 10de 1467)",
+		)
 	}
 
 	return nil
@@ -65,10 +67,13 @@ func root(cmd *cobra.Command, args []string) error {
 	var ids []PCIID
 	var err error
 
-	if len(args) == numSubDeviceIDs {
+	switch len(args) {
+	case numSubDeviceIDs:
 		ids, err = QuerySubDevice(args[0], args[1], args[2], args[3])
-	} else {
+	case numDeviceIDs:
 		ids, err = QueryDevice(args[0], args[1])
+	case numVendorIDs:
+		ids, err = QueryVendor(args[0])
 	}
 
 	if err != nil {

--- a/main.go
+++ b/main.go
@@ -13,6 +13,7 @@ import (
 
 const (
 	version         = "v2.0.0"
+	numVendorIDs    = 1
 	numDeviceIDs    = 2
 	numSubDeviceIDs = 4
 	remoteURL       = "https://raw.githubusercontent.com/pciutils/pciids/master/pci.ids"
@@ -67,6 +68,29 @@ func LatestWithContext(ctx context.Context) (string, error) {
 	return string(bodyBytes), nil
 }
 
+func QueryVendor(vendorID string) ([]PCIID, error) {
+	results := make([]PCIID, 0)
+
+	vID := strings.ToLower(vendorID)
+
+	log.Debug(fmt.Sprintf("looking up %s", vID))
+
+	pciids, err := All()
+	if err != nil {
+		return results, err
+	}
+
+	for _, pciid := range pciids {
+		if vID == pciid.VendorID {
+			results = append(results, pciid)
+		}
+	}
+
+	log.Debug(fmt.Sprintf("found %d result(s)", len(results)))
+
+	return results, nil
+}
+
 // Device finds all matching devices based on a pair of vendor and device IDs.
 func QueryDevice(vendorID string, deviceID string) ([]PCIID, error) {
 	results := make([]PCIID, 0)
@@ -101,7 +125,8 @@ func QueryDevice(vendorID string, deviceID string) ([]PCIID, error) {
 // SubDevice finds all matching devices based on a quartet of IDs.
 func QuerySubDevice(
 	vendorID string, deviceID string, subVendorID string, subDeviceID string) (
-	[]PCIID, error) {
+	[]PCIID, error,
+) {
 	results := make([]PCIID, 0)
 
 	vID := strings.ToLower(vendorID)

